js/src/filetree.ts

```diff
@@ -260,10 +260,13 @@ export class FileTree extends Widget {
         icon = this.dr.getFileType("text").icon.element({
           className: "jp-DirListing-itemIcon",
         });
-      } else {
+      } else if (fileTypes[0].icon) {
```

Comment:
do we need to range check on `fileTypes` or are we sure its got `length > 0`

Reply:
as i recall yes we are sure `length > 0` here. i'll double check

Reply:
The condition in the [preceding `if` clause](https://github.com/jpmorganchase/jupyter-fs/blob/d2ef8f3f0f924dfbfba57461287c6cca6bc51238/js/src/filetree.ts#L259) is `fileTypes.length === 0`, so yeah, it's legit here to assume `length > 0`

Reply:
Should these changes propagate to James's main repo?

Reply:
yep, same issue with not treating `.icon` as optional is [right here in jupyterlab_filetree](https://github.com/youngthejames/jupyterlab_filetree/blob/7e14cc84e66305e598e18b6ae432c18cb8b7352e/src/filetree.ts#L227-L233). I'll look into it